ADALiOS/ADALiOS/ADProfileInfo.m

```diff
@@ -0,0 +1,269 @@
+// Copyright © Microsoft Open Technologies, Inc.
+//
+// All Rights Reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+// ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A
+// PARTICULAR PURPOSE, MERCHANTABILITY OR NON-INFRINGEMENT.
+//
+// See the Apache License, Version 2.0 for the specific language
+// governing permissions and limitations under the License.
+
+#import "ADProfileInfo.h"
+#import "ADALiOS.h"
+#import "ADOAuth2Constants.h"
+#import "NSString+ADHelperMethods.h"
+
+static NSString* const PROFILE_INFO_SUBJECT = @"sub";
+static NSString* const PROFILE_INFO_OBJID = @"oid";
+static NSString* const PROFILE_INFO_TENANTID = @"tid";
+static NSString* const PROFILE_INFO_PREFERRED_USERNAME = @"preferred_username";
+static NSString* const PROFILE_INFO_FRIENDLY_NAME = @"name";
+static NSString* const PROFILE_INFO_TYPE = @"typ";
+static NSString* const PROFILE_INFO_JWT_TYPE = @"JWT";
+
```

Comment:
You should not drop claims from the server randomly. Everything should be persisted

Sent from Outlook<http://aka.ms/Ox5hz3>




On Wed, Sep 16, 2015 at 4:15 PM -0700, "Ryan Pangrle" <notifications@github.com<mailto:notifications@github.com>> wrote:


In ADALiOS/ADALiOS/ADProfileInfo.m<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fAzureAD%2fazure-activedirectory-library-for-objc%2fpull%2f378%23discussion_r39696880&data=01%7c01%7cKanishk.Panwar%40microsoft.com%7c30c8225edc2c4a25413c08d2beeca04a%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=Gcokm6FG11eDPeZ8D2zBbgY17oNR24JhIAoG9h2bobk%3d>:

> +// See the Apache License, Version 2.0 for the specific language
> +// governing permissions and limitations under the License.
> +
> +#import "ADProfileInfo.h"
> +#import "ADALiOS.h"
> +#import "ADOAuth2Constants.h"
> +#import "NSString+ADHelperMethods.h"
> +
> +static NSString* const PROFILE_INFO_SUBJECT = @"sub";
> +static NSString* const PROFILE_INFO_OBJID = @"oid";
> +static NSString* const PROFILE_INFO_TENANTID = @"tid";
> +static NSString* const PROFILE_INFO_PREFERRED_USERNAME = @"preferred_username";
> +static NSString* const PROFILE_INFO_FRIENDLY_NAME = @"name";
> +static NSString* const PROFILE_INFO_TYPE = @"typ";
> +static NSString* const PROFILE_INFO_JWT_TYPE = @"JWT";
> +


Is there a particular reason why "ver" is interesting, or worth caching?

-
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fAzureAD%2fazure-activedirectory-library-for-objc%2fpull%2f378%2ffiles%23r39696880&data=01%7c01%7cKanishk.Panwar%40microsoft.com%7c30c8225edc2c4a25413c08d2beeca04a%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=e81H7jusa3XSK8A6knfvO3RT7o2sCRgoNOqdwydLgfc%3d>.